### PR TITLE
Make `GetTxnScopeFromConfig` return `TxnScope` from config directly

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -139,24 +142,23 @@ func UpdateGlobal(f func(conf *Config)) func() {
...
 		if len(v) > 0 {
 			return false, v
 		}
-		return true, globalTxnScope
+		return true, oracle.GlobalTxnScope
 	}
 
 	if kvcfg := GetGlobalConfig(); kvcfg != nil && len(kvcfg.TxnScope) > 0 {
-		return false, kvcfg.TxnScope
+		if kvcfg.EnableLocalTxn {
```
Comment (Xuanwo) [inline config/config.go:156]: How about check `kvcfg.EnableLocalTxn && len(kvcfg.TxnScope) > 0` at the same level? That will looks more readable to me.

**Review by Xuanwo — APPROVED** [review]

File: config/config.go [hunk 2]
```diff
@@ -139,24 +142,28 @@ func UpdateGlobal(f func(conf *Config)) func() {
...
 	}
 
-	if kvcfg := GetGlobalConfig(); kvcfg != nil && len(kvcfg.TxnScope) > 0 {
-		return false, kvcfg.TxnScope
+	// If EnableLocalTxn is false, we will always return it's global.
+	if enableLocalTxn && len(txnScope) > 0 {
+		return false, txnScope
 	}
-	return true, globalTxnScope
+	return true, oracle.GlobalTxnScope
```
Comment (nolouch) [inline config/config.go:166]: Does this affect the stale read? https://github.com/pingcap/tidb/pull/25525/files#diff-ed09db7ac6827940f6b3ac9d72565c4fea1136d8e6ff76ad7aa7ffefc73aca26R1535

Comment (nolouch) [reply config/config.go:166]: cc @Yisaer

**Review by nolouch — APPROVED** [review]

**Comment by JmPotato** [comment]
The integration failed because of the change of `GetTxnScopeFromConfig` function signature. This will be fixed after merging pingcap/tidb#25559.

**Review by nolouch — APPROVED** [review]
lgtm

**Review by Yisaer — APPROVED** [review]

**Pull request merged**